src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -37,9 +38,8 @@ public Emitter(Compilation compilation, bool emitPreamble = true) : base(emitPre
             else
             {
                 _modifier = "internal";
-                string suffix = $"_{new Random().Next():X8}";
```

Comment:
>string suffix = $"_{new Random().Next():X8}"; [](http://example.com/codeflow?start=16&length=45)

Eirik suggested to change this for the configuration source gen, but we need to keep it using random here as we need to avoid cross assembly names when assemblies marked with InternalsVisibleToAttribute. If this is the case for configuration too, then we should stick with random everywhere.

Reply:
https://github.com/dotnet/runtime/pull/90340#discussion_r1297638177. It's doing the same thing, just sharing the code with config binding gen.